resources/assets/js/components/ApplicantProfile/Skills/ProfileSkillsPage.tsx

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { FormattedMessage, useIntl } from "react-intl";
+import { getId, notEmpty, toIdMap } from "../../../helpers/queries";
+import {
+  useApplicantExperience,
+  useApplicantExperienceSkills,
+  useApplicantSkillIds,
+  useSkillCategories,
+  useSkills,
+} from "../../../hooks/apiResourceHooks";
+import { Experience, ExperienceSkill, Skill } from "../../../models/types";
+import { loadingMessages } from "../../Application/applicationMessages";
+import FindSkillsModal, { FindSkillsModalTrigger } from "../../FindSkillsModal";
+import RootContainer from "../../RootContainer";
+import List from "./List";
+
+export const ProfileExperiencePage: React.FC<{ applicantId: number }> = ({
+  applicantId,
+}) => {
+  const intl = useIntl();
+
+  const skillsResource = useSkills();
+  const skillCategoriesResource = useSkillCategories();
+  const applicantSkillsResource = useApplicantSkillIds(applicantId);
+  const experienceResource = useApplicantExperience(applicantId);
+  const experienceSkillResource = useApplicantExperienceSkills(applicantId);
+
+  const idToSkill = toIdMap(skillsResource.value);
+  const applicantSkills = applicantSkillsResource.value.skill_ids
+    .map((skillId) => idToSkill.get(skillId))
+    .filter(notEmpty);
+  const experiences: Experience[] = Object.values(experienceResource.values);
+  const experienceSkills: ExperienceSkill[] = Object.values(
+    experienceSkillResource.values,
+  );
+
+  const submitNewSkills = async (newSkills: Skill[]): Promise<void> => {
+    await applicantSkillsResource.update({
+      skill_ids: [
+        ...applicantSkillsResource.value.skill_ids,
+        ...newSkills.map(getId),
+      ],
+    });
+  };
+
+  const removeSkill = async (skillId: number): Promise<void> => {
```

Comment:
This block of code is likely not the culprit, but I found an edge case where if there is only one skill attached to the applicant, it cannot be removed (422 Unprocessable Entity response is returned, see screenshot).

![422 Unprocessable Entity response is returned](https://user-images.githubusercontent.com/3046459/109043965-d88e1580-769f-11eb-9019-fe413141acfb.png)

Reply:
Nice catch. Had to change the backend validation from `required` to `present`.